torch/fx/graph_module.py

```diff
@@ -137,6 +138,18 @@ def __new__(cls: 'Type[GraphModule]', *args, **kwargs):
 
         class GraphModuleImpl(cls):  # type: ignore
             pass
+
+        # If possible, change the calling function name to match the
+        # user's original code for better error reporting
+        parent_frame = inspect.currentframe().f_back
+        calling_name = cls.__name__
+        if "root" in parent_frame.f_locals:
+            calling_name = parent_frame.f_locals["root"].__class__.__name__
```

Comment:
This is too brittle; It's making assumptions about the caller and breaks the procedural abstraction. Callees should not assume anything about the caller beyond the contract established by the function signature

I think we should instead do the following:

1. Here in `__new__`, set `GraphModuleImpl.__name__ = 'GraphModule'`. This will be the default value
2. Add a parameter `class_name : Optional[str] = None` to the `GraphModule` constructor. Internally, the constructor will set `self.__class__.__name__ = class_name` if that name is not None
3. `symbolic_trace` will pass the value of `root.__class__.__name__` for this parameter

Note that the case where we want to populate the `class_name` parameter is limited to `symbolic_trace`. In other instances of creating a `GraphModule`, it's the caller's (i.e. some pass writer's) responsibility to provide a name that makes sense